### ISSUE TITLE
ServicePoint retrieved via FindServicePoint does not contain server certificate

### ISSUE THREAD

**Issue by bojanrajkovic** [dotnet/corefx#26535]
A typical way to obtain the server certificate after making a request via `HttpClient` is to use `ServicePointManager.FindServicePoint` with the URI and grab the `Certificate` property off of that.

This works fine in .NET 4.5, as evidenced by this fragment from a [workbook]:

![image](https://user-images.githubusercontent.com/110747/35256633-b8f1174c-ffc2-11e7-8793-629517806b68.png)

However, in a .NET Core application, this doesn't work:

![image](https://user-images.githubusercontent.com/110747/35256704-10a6de18-ffc3-11e7-99d5-67c4db540a3b.png)

You can also see the attached project in [certificate-test.zip](https://github.com/dotnet/corefx/files/1654414/certificate-test.zip). It contains a sample you can download and `dotnet run`.

[workbook]: https://github.com/Microsoft/workbooks

**Comment by bojanrajkovic** [comment]
Even trying to retrieve the `X509Certificate`/`X509Certificate2` via a `ServerCertificateCustomValidationCallback` results in an invalid certificate. See the following:

![image](https://user-images.githubusercontent.com/110747/35257489-9b9643de-ffc7-11e7-96b9-3df6afdf3ada.png)

Something seems to be very broken with certificates in .NET Core.

**Comment by bojanrajkovic** [comment]
What _does_ work is creating a `TcpClient`, wrapping the stream in an `SslStream`, calling `AuthenticateAsClient`, and then checking the `RemoteCertificate` property.

**Comment by davidsh** [comment]
ServicePoint is not supported on .NET Core.   If you need to grab the server certificate using HttpClient, you can use the ServerCertificateCustomValidationCallback in HttpClientHandler:

https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.servercertificatecustomvalidationcallback?view=netcore-2.0

The server certificate is passed into this callback.

cc: @karelz @geoffkizer

**Comment by bojanrajkovic** [comment]
Yep, I figured that out, but it would be good to either document this, or make the ServicePoint-related code throw not supported exceptions on .NET Core. As it is, it seemed to be in a half-working state, because other parts of the SP _were_ properly set, just not the certificate.

**Comment by karelz** [comment]
I guess we could document that it is deprecated / compat-only on https://github.com/dotnet/platform-compat/issues

**Comment by karelz** [comment]
Triage: We need the documentation. Mention alternative: `HttpClient` with certificate validation callback.
